src/pages/atomicdex/api/v20-dev/non_fungible_tokens/index.mdx

```diff
@@ -0,0 +1,577 @@
+    ],
+    "url": "https://moralis-proxy.komodo.earth"
+  }
+}
+```
+</CodeGroup>
+
+
+### Response Parameters
+
+| Parameter            | Type             | Description                                                                                      |
+| ---------------------| ---------------- | ------------------------------------------------------------------------------------------------ |
+| amount               | string           | Amount of tokens.                                                                                |
+| block_number_minted  | integer          | Block height when the token was minted.                                                          |
+| block_number         | integer          | Current block height.                                                                            |
+| chain                | string           | Chain name. One of `AVALANCHE`, `BSC`, `ETH`, `FANTOM`, or `POLYGON`.                            |
+| collection_name      | string           | Collection name. May be `null`.                                                                  |
+| contract_type        | string           | Contract type. `ERC721` or `ERC1155`.                                                            |
+| owner_of             | string           | Address of the token owner.                                                                      |
+| token_address        | string           | Token address.                                                                                   |
+| token_id             | string           | Token ID.                                                                                        |
+| token_hash           | string           | Token hash.                                                                                      |
+| symbol               | string           | Token symbol.                                                                                    |
+| token_uri            | string           | Token URI.                                                                                       |
+| metadata             | string           | Token metadata.                                                                                  |
+| last_token_uri_sync  | string           | Last time the token URI was synced.                                                              |
+| last_metadata_sync   | string           | Last time the token metadata was synced.                                                         |
+| minter_address       | string           | Minter address.                                                                                  |
+| possible_spam        | boolean          | Possible spam.                                                                                   |
+| uri_meta             | object           | A Standard [TokenMetadata](LINK TO COMMON STRUCTURE) object.                                     |
```

Comment:
This is how Nft structure currently looks like in [add-token-uri-info-to-nft-struct ](https://github.com/KomodoPlatform/atomicDEX-API/tree/add-token-uri-info-to-nft-struct) branch. This is not in dev yet, but GUI team uses this branch bcz they need `UriMeta`
you can check which fields can be null
```rust
pub struct Nft {
    pub(crate) chain: Chain,
    pub(crate) token_address: String,
    pub(crate) token_id: BigDecimal,
    pub(crate) amount: BigDecimal,
    pub(crate) owner_of: String,
    pub(crate) token_hash: String,
    pub(crate) block_number_minted: u64,
    pub(crate) block_number: u64,
    pub(crate) contract_type: Option<ContractType>,
    pub(crate) collection_name: Option<String>,
    pub(crate) symbol: Option<String>,
    pub(crate) token_uri: Option<String>,
    pub(crate) metadata: Option<String>,
    pub(crate) last_token_uri_sync: Option<String>,
    pub(crate) last_metadata_sync: Option<String>,
    pub(crate) minter_address: Option<String>,
    pub(crate) possible_spam: Option<bool>,
    pub(crate) uri_meta: UriMeta,
}
```
```rust
pub(crate) struct UriMeta {
    image: Option<String>,
    #[serde(rename(deserialize = "name"))]
    token_name: Option<String>,
    description: Option<String>,
    attributes: Option<Json>,
    animation_url: Option<String>,
}
```
ContractType is enum and yeah it could be ERC721 or ERC1155 (uppercase)

Reply:
Updated, using descriptions from https://docs.moralis.io/web3-data-api/evm/reference/get-wallet-nfts?address=0xd8da6bf26964af9d7eed9e03e53415d37aa96045&chain=eth&format=decimal&token_addresses=[]&media_items=false